lib/imap.php

```diff
@@ -85,12 +85,19 @@ public function checkPassword($uid, $password) {
 					$groups[] = $pieces[1];
 		}
 
-		$protocol = ($this->sslmode === "ssl") ? "imaps" : "imap";
-		$url = "{$protocol}://{$this->mailbox}:{$this->port}";
 		$ch = curl_init();
-		if ($this->sslmode === 'tls') {
-			curl_setopt($ch, CURLOPT_USE_SSL, CURLUSESSL_ALL);
+
+		if ($this->sslmode !== 'tls') {
+			$protocol = 'imaps';
+		    // Use STARTTLS as default encryption mode
+			if ($this->sslmode) {
+				curl_setopt($ch, CURLOPT_USE_SSL, CURLUSESSL_ALL);
+			}
+		} else {
+			$protocol = 'imaps';
```

Comment:
you set the `$protocol` two times to `imaps` but for starttls (and insecure connections) we would need `imap`, right?